include/fcl/narrowphase/detail/convexity_based_algorithm/gjk_libccd-inl.h

```diff
@@ -1005,6 +1005,130 @@ static int __ccdEPA(const void *obj1, const void *obj2,
+      if(p2) *p2 = simplex->ps[simplex->last].v2;
+  }
+  else if (simplex_size ==2)
+  {
+      // Closest points lie on the segment defined by the points in the simplex
+      // Let the segment be defined by points A and B. We can write p as
+      //
+      // p = A + s*AB, 0 <= s <= 1
+      // p - A = s*AB
+      //
+      // This defines three equations, but we only need one. Take the x
+      // component
+      //
+      // p_x - A_x = s*AB_x
+      //
+      // Thus, s is given by
+      //
+      // s = (p_x - A_x)/(B_x - vA_x)
+      double A_x{ccdVec3X(&(simplex->ps[0].v))};
+      double B_x{ccdVec3X(&(simplex->ps[1].v))};
+      double p_x{ccdVec3X(p)};
+      double    s{(p_x - A_x) / (B_x - A_x)};
+      if (p1)
+      {
+        // p1 = A1 + s*A1B1
+        ccd_vec3_t sAB;
+        ccdVec3Sub2(&sAB, &(simplex->ps[1].v1), &(simplex->ps[0].v1));
+        ccdVec3Scale(&sAB, s);
+        ccdVec3Copy(p1, &(simplex->ps[0].v1));
+        ccdVec3Add(p1, &sAB);
+      }
+      if (p2)
+      {
+        // p2 = A2 + s*A2B2
+        ccd_vec3_t sAB;
+        ccdVec3Sub2(&sAB, &(simplex->ps[1].v2), &(simplex->ps[0].v2));
+        ccdVec3Scale(&sAB, s);
+        ccdVec3Copy(p2, &(simplex->ps[0].v2));
+        ccdVec3Add(p2, &sAB);
+      }
+  }
+  else // simplex_size == 3
+  {
+    // Closest points lie in the triangle defined by the points in the simplex
+    ccd_vec3_t AB, AC, n, v_prime, AB_cross_v_prime, AC_cross_v_prime, AB_cross_AC;
+    // Let the triangle be defined by points A, B, and C. The triangle lies in
+    // the plane that passes through A, whose normal is given by
+    //
+    // n = (AB x AC) / || AB x AC ||
+    ccdVec3Sub2(&AB, &(simplex->ps[1].v), &(simplex->ps[0].v));
+    ccdVec3Sub2(&AC, &(simplex->ps[2].v), &(simplex->ps[0].v));
+    ccdVec3Cross(&n, &AB, &AC);
+    ccdVec3Normalize(&n);
+
+    // Since p lies in ABC, it can be expressed as
+    //
+    // p = A + p'
+    // p' = s*AB + t*AC
+    //
+    // where 0 <= s, t, s+t <= 1.
+    ccdVec3Sub2(&v_prime, p, &(simplex->ps[0].v));
+
+    // To find the corresponding v1 and v2, we need
+    // values for s and t. Taking cross products with AB and AC gives the
+    // following system:
+    //
+    // AB x p' =  t*(AB x AC)
+    // AC x p' = -s*(AB x AC)
+    ccdVec3Cross(&AB_cross_v_prime, &AB, &v_prime);
+    ccdVec3Cross(&AC_cross_v_prime, &AC, &v_prime);
+    ccdVec3Cross(&AB_cross_AC, &AB, &AC);
+
+    // To convert this to a system of scalar equations, we take the dot product
+    // with n:
+    //
+    // n . (AB x p') =  t * n . (AB x AC)
+    // n . (AC x p') = -s * n . (AB x AC)
+    double n_dot_AB_cross_AC{ccdVec3Dot(&n, &AB_cross_AC)};
+
+    // Therefore, s and t are given by
+    //
+    // s = -n . (AC x p') / n . (AB x AC)
+    // t =  n . (AB x p') / n . (AB x AC)
+    double s{-ccdVec3Dot(&n, &AC_cross_v_prime) / n_dot_AB_cross_AC};
+    double t{ccdVec3Dot(&n, &AB_cross_v_prime) / n_dot_AB_cross_AC};
+
```

Comment:
I believe a little algebraic manipulation simplifies the math (making it cheaper and more robust -- no square roots and fewer floating point ops).

Starting with the following definitions (and focusing only on `t` with the understanding that it applies directly to `s`:
```
n = AB x AC
n̂ = n / ||n||
t = n̂⋅(AC x p') / (n̂⋅(AB x AC))
```
We get the following:
```
t = n̂⋅(AC x p') / (n̂⋅(AB x AC))                              - the given equation
t = (n / ||n||)⋅(AC x p') / ((n / ||n||)⋅(AB x AC))          - substitute for n̂
t = n⋅(AC x p') / (n⋅(AB x AC))                              - cancel 1 / ||n||
t = (AB x AC)⋅(AC x p') / ((AB x AC)⋅(AB x AC))              - substitute for n
t = (AB x AC)⋅(AC x p') / ||AB x AC||²                       - simplify
```